core/admin/mailu/utils.py

```diff
@@ -408,23 +397,12 @@ def save_session(self, app, session, response):
         if session.accessed:
             response.vary.add('Cookie')
 
-        set_cookie = session.permanent and app.config['SESSION_REFRESH_EACH_REQUEST']
-        need_refresh = session.needs_refresh()
-
-        # save modified session or refresh unmodified session
-        if session.modified or need_refresh:
-            set_cookie |= session.save()
-
-        # set cookie on refreshed permanent sessions
-        if need_refresh and session.permanent:
-            set_cookie = True
-
-        # set or update cookie if necessary
-        if set_cookie:
+        # save session and update cookie if necessary
+        if session.save():
```

Comment:
Why did you revert back to "save&refresh on every request"?
I did this intentionally after SESSION_TIMEOUT/2 (half the session time).